### PR TITLE
SubMat needed for multi dimensional blobs from DNN.

### PR DESCRIPTION
**Pull request by TWhidden**
Needed the SubMat for OpenCvSharp, but was not implemented.  Looked at the c sig, and you already had the impl for it. Tested this and confirmed working. Needed for DNN batching. Example usage:

```
if (batchSize > 1)
{
    // get the sample slice from 3D matrix (batch, box, classes+5)
    var range = new[]
    {
        new Range(b, b + 1),
        Range.All,
        Range.All
    };

    var predOut = detections[i];
    outMat = predOut[range];
    var predOutSize = predOut.Size(1);
    outMat = outMat.Reshape(1, predOutSize);
}
else
{
    outMat = detections[i];
}
```

### CODE REVIEW

File: src/OpenCvSharp/Modules/core/Mat/Mat.cs [hunk 1]
```diff
@@ -3552,21 +3552,14 @@ public Mat SubMat(Rect roi)
...
-            for (int i = 0; i < ranges.Length; i++)
-            {
-                slices[i] = ranges[i];
-            }
 
-            IntPtr retPtr = NativeMethods.core_Mat_subMat1(ptr, ranges.Length, ranges);
+            IntPtr retPtr = NativeMethods.core_Mat_subMat2(ptr, ranges.Length, ranges);
             Mat retVal = new Mat(retPtr);
-            return retVal;*/
+            return retVal;
```
Comment (shimat) [inline src/OpenCvSharp/Modules/core/Mat/Mat.cs:3563]:
```suggestion
            GC.KeepAlive(this);
            return retVal;
```

Comment (TWhidden) [reply src/OpenCvSharp/Modules/core/Mat/Mat.cs:3563]: Looks good. Thanks!

Comment (shimat) [reply src/OpenCvSharp/Modules/core/Mat/Mat.cs:3563]: Cloud you press the "Commit suggestion" button to update your fork repository?
I don't have permission to push to TWhidden:master.

**Comment by shimat** [comment]
Thanks!

**Pull request merged**